superset/connectors/sqla/models.py

```diff
@@ -1073,8 +1083,34 @@ def mutator(df: pd.DataFrame) -> None:
     def get_sqla_table_object(self) -> Table:
         return self.database.get_table(self.table_name, schema=self.schema)
 
-    def fetch_metadata(self, commit=True) -> None:
+    def fetch_metadata(
+        self,
+        commit=True,
+        dttm_config: Optional[Dict[str, Dict[str, str]]] = None,
+        main_dttm_col: Optional[str] = None,
+    ) -> None:
```

Comment:
Is there some reason why the signature is modified? It appears we could just as easily pick up the config params here instead of passing them from `views.py`

Reply:
@villebro I have a slight preference towards a very lightweight model classes and not to depend on the superset app. A middle ground could be here to move this function outside of the models